### PR TITLE
Milestone 5 - React Calculator

### PR DESCRIPTION
**Pull request by JohnHernCode**
Milestones completed:
1. Initialize the state in the constructor with three properties: total, next and operation
2. By default, they should all be set to null.
3. Implement render function that should display Display and ButtonPanel.
4. Implement the handleClick function which takes name (in place of buttonName) as an argument and changes the state based on the result of the calculate function.
5. Pass the total or next value to Display to display the current result.
6. Pass handleClick as a clickHandler property to ButtonPanel
7. Implement the handleClick function which takes name (in place of buttonName) as an argument and returns clickHandler from props.
8. Pass the clickHandler property to each button in the panel.
9. Implement the handleClick function which takes name (in place of buttonName) as an argument and returns clickHandler from props.
10. Use the onClick event to attach the handleClick event to the button.

I decided not to worry about the styling, it was the actual functionality I was worried about, not how it looks. I can work on that later. 

NOTE: I used name, instead of buttonName. It works exactly the same.

### CODE REVIEW

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi John!
Great job so far:clap::clap:.**

**Major highlights**

- The PR is very professional
- The readme is also very descriptive
- The codebase is clean, easy to read and understand.

However, we still need to make a few changes to get the project approved:muscle:.

**To improve**

- [x] ButtonPanel component

- it implements `handleClick` function ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/implementHandleClick.png)). Kindly implement this function inside this component.

Make the above-required changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Comment by JohnHernCode** [comment]
Are you absolutely sure it is supposed to be in the button panel? It is only working correctly in my app.js file where it originally was.. 

 handleClick(buttonName) {
    const oldState = this.state;
    this.setState(calculate(oldState, buttonName));
  }

I even have it binded there in app.js ... 

It doesn't seem logical to have it in the button panel. 


Maybe I am totally seeing this wrong... any advice?

File: src/components/App.js [hunk 1]
```diff
@@ -11,15 +11,18 @@ class App extends Component {
...
-      // eslint-disable-next-line react/no-unused-state
       operation: null,
     };
     this.handleClick = this.handleClick.bind(this);
   }
 
-  handleClick(buttonName) {
-    const oldState = this.state;
-    this.setState(calculate(oldState, buttonName));
+  handleClick = (name) => {
```
Comment (arslanbisharat) [inline src/components/App.js]: - [x] Thank you for the explanation but you should implement that change As requested by previous CR.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested , 

Hi @JohnHernCode, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/App.js (hunk 1 above)
Comment (JohnHernCode) [reply src/components/App.js]: Oh I get it, I was somehow just blocked from seeing the changes I needed to make.

Comment (JohnHernCode) [reply src/components/App.js]: I used name for everything instead of buttonName

I had gone through everything and changed it all.

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**